bigchaindb/web/views/transactions.py

```diff
@@ -55,6 +55,12 @@ def post(self):
         Return:
             A ``dict`` containing the data about the transaction.
         """
+        parser = reqparse.RequestParser()
+        parser.add_argument('mode', type=parameters.valid_mode)
+        args = parser.parse_args()
+        if not str(args['mode']) or str(args['mode']) == 'None':
```

Comment:
Small query, could these conditions be handled in `valid_mode` function?

Reply:
Actually no. 'None' is the default value, so the first case (empty mode) should actually never exist. The second check is for the default value which is set by flask. Since it is set by flask there is no need for validation. But I'll change it. I didn't think of it but I can just change the default value and we don't need this line at all.

Reply:
I couldn't understand you, what I was asking is if following is possible
```python
def valid_mode(mode):
    if mode == 'async' or mode == None or mode == '':
        return 'broadcast_tx_async'
    if mode == 'sync':
        return 'broadcast_tx_sync'
    if mode == 'commit':
        return 'broadcast_tx_commit'
    raise ValueError('Mode must be "async", "sync" or "commit"')
```

Reply:
No. As far as I understand it when we don't set a mode the mode will be set to a default. In this case, it will never be `None`. Furthermore, I changed it now, so the default is always `'broadcast_tx_async'` instead of `'None'`. 
If I revert the changes and the default will be `'None'` it then is set by Flask and therefore doesn't enter the validation. 
I hope that explains it.

Reply:
Yes, I think I understand it now.